crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -193,7 +193,12 @@ fn insert_reflect(
     let Some(mut entity) = world.get_entity_mut(entity) else {
         panic!("error[B0003]: Could not insert a reflected component (of type {}) for entity {entity:?} because it doesn't exist in this World.", component.reflect_type_path());
     };
-    let Some(type_registration) = type_registry.get_with_type_path(type_info) else {
+    let Some(type_registration) = type_registry.get_with_type_path(type_info).or_else(|| {
```

Comment:
I would suggest to just use `get_represented_type_info` in line 192.

Reply:
After a bench I found that it doesn't make much influence in performance using `get_represented_type_info` in this case. Cloning costs time much more.
For 10k iters
No cloning: `reflect_type` 22 μs, `get_represented_type_info` 39 μs
dyn clone: `reflect_type` 365 μs, `get_represented_type_info` 379 μs
clone_value: 1,177 μs

It should be OK just using `get_represented_type_info`.